src/appleseed/renderer/modeling/bssrdf/sss.cpp

```diff
@@ -164,8 +173,120 @@ double gaussian_profile_pdf(
 // Dipole diffusion profile implementation.
 //
 
+//
+// Sampling derivation:
+//
+//   We want to generate samples according to R(r) * 2 * Pi * r.
+//   We choose the exponential term in all dipole formulations:
+//
+//     exp(-sigma_tr * r)
+//
+//   Our distribution is:
+//
+//     p(r) = exp(-sigma_tr * r) * 2 * Pi * r
+//
+//   And the normalization factor is:
+//
+//     / inf
+//     |                 2 * Pi
+//     |  p(r) dr  =  ------------
+//     |               sigma_tr^2
+//     / 0
+//
+//  Our PDF is:
+//
+//    pdf(r) = sigma_tr^2 * exp(-sigma_tr * r) * r
+//
+//  We integrate the PDF to find the CDF:
+//
+//     / R
+//     |
+//     |  pdf(r) dr  = 1 - exp(-sigma_tr * r) * (1 + sigma_tr * r)
+//     |
+//     / 0
+//
+//  The CDF is not analytically invertible.
+//  We use a combination of bisection and Newton-Raphson steps
+//  to invert it numerically.
+//
+
+double dipole_pdf(const double r, const double sigma_tr)
+{
+    return square(sigma_tr) * exp(-sigma_tr * r) * r;
+}
+
+double dipole_cdf(const double r, const double sigma_tr)
+{
+    const double x = sigma_tr * r;
+    return 1.0 - exp(-x) * (1.0 + x);
+}
+
+namespace
+{
+
+struct DipoleCDFFun
```

Comment:
Indent content of namespace.

Only the content of top-level namespaces is not indented since this would indent the whole file without adding any value.